akka-cluster/src/main/scala/akka/cluster/routing/ClusterRouterConfig.scala

```diff
@@ -46,33 +55,46 @@ final case class ClusterRouterGroupSettings(
   totalInstances:    Int,
   routeesPaths:      immutable.Seq[String],
   allowLocalRoutees: Boolean,
-  useRole:           Option[String]) extends ClusterRouterSettingsBase {
+  useRoles:          Set[String]) extends ClusterRouterSettingsBase {
```

Comment:
In `akka.cluster.UniqueAddress` we have done something like this, changing the type of a field, that could be a good thing to look at.